### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by nanu-c**
# Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Description

**IMPORTANT TO KNOW**

**CHANGES**

This correctly creates the keycloak groups and improves the logs for failed payments.

**TODO**

# Checklist:

- [ ] I have commented my code (or ChatGPT did), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings, neither in my IDE nor in my browser
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/keycloak/keycloak.go [hunk 1]
```diff
@@ -195,32 +202,39 @@ func (k *Keycloak) AssignRole(userID string, roleName string) error {
 // Assign group to user
 func (k *Keycloak) AssignGroup(userID string, groupName string) error {
 	k.checkAdminToken()
+	if groupName[0] != '/' {
+		groupName = "/" + groupName
+	}
```
Comment (lebe1) [inline app/keycloak/keycloak.go:207]: Why did you add this if-clause? Maybe add a comment here for explainability

**Review by lebe1 — COMMENTED** [review]
Just one small suggestion to provide at least one comment for more explainability in the code. Everything else looks good especially a lot of the error logs are a proper improvement!

File: app/keycloak/keycloak.go (hunk 1 above)
Comment (nanu-c) [reply app/keycloak/keycloak.go:207]: Groups can only be searched by group paths and not my group names. Group paths have to start with `/` and this is a check for it.
